_posts/2020-02-07-metal3-dev-env-install-deep-dive.md

```diff
@@ -0,0 +1,921 @@
+* `Container registry` where the virtual servers will pull the images needed to run a K8s installation.
+<br>
+<br>
+> **Information:** A baseboard management controller (BMC) is a specialized service processor that monitors the physical state of a computer, network server or other hardware device using sensors and communicating with the system administrator through an independent connection. The BMC is part of the Intelligent Platform Management Interface (IPMI) and is usually contained in the motherboard or main circuit board of the device to be monitored.
+<br>
+<br>
+
+First, an ssh-key in charge of communicating to libvirt is created if it does not exist previously. This key is called `id_rsa_virt_power`. It is added to the root authorized_keys and is used by the `vbmc` and `sushy tools` to contact libvirt.
+<br>
+<br>
+> **Information:** `sushy-tools` is a set of simple simulation tools aiming at supporting the development and testing of the Redfish protocol implementations.
+<br>
+<br>
+
+Next, another Ansible playbook called [setup-playbook.yml](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/setup-playbook.yml) is run against the host. It is focused on set up the virtual infrastructure around `metal3-dev-env`. Below it is shown the Ansible variables that are passed to the playbook, which actually are obtaining the values from the global variables defined in the [common.sh](#commonsh) or the configuration file.
+
+```sh
+ANSIBLE_FORCE_COLOR=true ansible-playbook \
+    -e "working_dir=$WORKING_DIR" \
+    -e "num_nodes=$NUM_NODES" \
+    -e "extradisks=$VM_EXTRADISKS" \
+    -e "virthost=$HOSTNAME" \
+    -e "platform=$NODES_PLATFORM" \
+    -e "libvirt_firmware=$LIBVIRT_FIRMWARE" \
+    -e "default_memory=$DEFAULT_HOSTS_MEMORY" \
+    -e "manage_baremetal=$MANAGE_BR_BRIDGE" \
+    -e "provisioning_url_host=$PROVISIONING_URL_HOST" \
+    -i vm-setup/inventory.ini \
+    -b -vvv vm-setup/setup-playbook.yml
+```
+
+| **Name of the variable** | **Default value**      |
+| ------------- | ---------- | 
+| WORKING_DIR | /opt/metal3-dev-env |
+| NUM_NODES | 2 |
+| VM_EXTRADISKS | false |
+| HOSTNAME | `hostname` |
+| NODES_PLATFORM | libvirt |
+| LIBVIRT_FIRMWARE | bios |
+| DEFAULT_HOSTS_MEMORY | 8192 |
+| MANAGE_BR_BRIDGE | y
+| PROVISIONING_URL_HOST | 172.22.0.1 |
+
+<br>
+> **Information:** There are variables that are only defined as Ansible variables, e.g. number of CPUs of the virtual bare metal server, size of disks, etc. In case you would like to change properties not defined globally in the `metal3-dev-env` take a look a the default variables specified in role: [common](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/common/defaults/main.yml) and [libvirt](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/libvirt/defaults/main.yml)
+<br>
+<br>
+
+
+
+The`setup-playbook.yml` is composed by 3 roles, which are detailed below:
+
+<br>
+
+> * **Common.**
```

Comment:
Hmmm... formatting issue? by using `>` here things get weird... it should be a simple list, right?